bootstrapper/internal/kubernetes/k8sapi/k8sutil.go

```diff
@@ -75,34 +78,30 @@ func NewKubernetesUtil() *KubernetesUtil {
 	}
 }
 
+// InstallComponentsFromCLI installs the kubernetes components passed from the CLI.
+func (k *KubernetesUtil) InstallComponentsFromCLI(ctx context.Context, kubernetesComponents versions.ComponentVersions) error {
+	for _, component := range kubernetesComponents {
+		if err := k.inst.Install(
+			ctx, component.URL, component.Hash, component.InstallPath, component.Extract,
+		); err != nil {
```

Comment:
Since we are trying to keep the number of function arguments at a reasonable level, would it make sense to just pass the `component` struct here, instead of pulling the values from it and passing them one by one?

Reply:
I also thought about that, but decided against it. My thought process was that we differentiate between `InstallComponents` and `Install` i.e. `Install` has a more general Interface. Also my question is if we should always optimize the number of parameters, or should we just stay below 5?

Reply:
>i.e. `Install` has a more general Interface

Does it need to be more general? 
And is it really more general if all of the parameters are just variables of `versions.ComponentVersions`?
IMO this isn't as much about keeping low number of arguments, but about knowing exactly what variables are needed for the function.

E.g. a function with the parameters `(context.Context, string, string, string, bool)` isn't as clear as `(context.Context, versions.ComponentVersions)`.
Especially since the later option allows us to set doc comments on the individual parts.